src/Agent.Worker/TestResults/FileZipper.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    internal class FileZipper : IZipper
```

Comment:
Same goes for IZipper, please add a ServierLocator, you can avoid MockZipper totally.
We do not have a pattern of creating mockable (or testable) class created. The problem is we create testclasses and implement its functionality. But another test would require the testclass to behave in a different way, then we start customizing these testclasses. All these can go away with if you move to servicelocatoer.

In every test which requires your interface implementation, create a mock and setup your required methods and register with testhostcontext.